pkg/security/security_profile/activity_tree/process_node_snapshot.go

```diff
@@ -152,39 +173,34 @@ func (pn *ProcessNode) snapshotFiles(p *process.Process, stats *Stats, newEvent
 
 		// TODO: add open flags by parsing `/proc/[pid]/fdinfo/fd` + O_RDONLY|O_CLOEXEC for the shared libs
 
-		_ = pn.InsertFileEvent(&evt.Open.File, evt, "", Snapshot, stats, false, reducer, nil)
+		_ = pn.InsertFileEvent(&evt.Open.File, evt, "", Snapshot, nil, false, reducer, nil)
 	}
 }
 
 // MaxMmapedFiles defines the max mmaped files
 const MaxMmapedFiles = 128
 
-func snapshotMemoryMappedFiles(pid int32, processEventPath string) ([]string, error) {
-	smapsPath := kernel.HostProc(strconv.Itoa(int(pid)), "smaps")
-	smapsFile, err := os.Open(smapsPath)
+func getMemoryMappedFiles(pid int32, processEventPath string) (files []string, _ error) {
```

Comment:
We cannot do that please keep the original code. This code will OOM kill in a lot of scenarii, mainly when the amount of memory mapped files is too big

Reply:
See https://github.com/DataDog/datadog-agent/pull/17393 and https://github.com/DataDog/datadog-agent/pull/23797 for what was done here before

Reply:
I kept the original code. Do you think upstream could use a walker instead of reading all the files ?

Reply:
Thanks ! I'm not 100% convinced by the move to named return value since we never use this in the rest of the codebase, but otherwise LGTM